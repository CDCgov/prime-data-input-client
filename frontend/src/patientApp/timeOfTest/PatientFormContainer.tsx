import React, { useEffect, useState } from "react";
import { connect, useSelector, useDispatch } from "react-redux";
import { useHistory, Redirect } from "react-router-dom";
import { toast } from "react-toastify";

import { setPatient as reduxSetPatient } from "../../app/store";
import { PxpApi } from "../../patientApp/PxpApiService";
import PersonForm from "../../app/patients/Components/PersonForm";
import PatientTimeOfTestContainer from "../PatientTimeOfTestContainer";
import { showNotification } from "../../app/utils";
import Alert from "../../app/commonComponents/Alert";

const PatientFormContainer = () => {
  const [prevPage, setPrevPage] = useState(false);
  const [nextPage, setNextPage] = useState(false);
  const patient = useSelector((state: any) => state.patient);
  const facility = useSelector((state: any) => state.facility);

  const dispatch = useDispatch();

  const plid = useSelector((state: any) => state.plid);
  const patientInStore = useSelector((state: any) => state.patient);

  useEffect(() => {
    window.scrollTo(0, 0);
  }, []);

  const history = useHistory();
  history.listen((_loc, action) => {
    if (action === "POP") {
      setPrevPage(true);
    }
  });

  if (prevPage) {
    return (
      <Redirect
        to={{
          pathname: "/patient-info-confirm",
        }}
      />
    );
  }

  if (nextPage) {
    return (
      <Redirect
        push
        to={{
          pathname: "/patient-info-symptoms",
        }}
      />
    );
  }

  const savePerson = async (person: Nullable<PersonFormData>) => {
    const {
      street,
      streetTwo,
      city,
      state,
      county,
      zipCode,
      residentCongregateSetting,
      employedInHealthcare,
      ...withoutAddress
    } = person;
    const updatedPatientFromApi = await PxpApi.updatePatient(
      plid,
      patientInStore.birthDate,
      {
        ...withoutAddress,
        residentCongregateSetting: person.residentCongregateSetting === "YES",
        employedInHealthcare: person.employedInHealthcare === "YES",
        address: {
          street: [street, streetTwo],
          city,
          state,
          county,
          zipCode,
        },
      }
    );
    showNotification(
      toast,
      <Alert type="success" title={`Your profile changes have been saved`} />
    );

    dispatch(
      reduxSetPatient({
        ...updatedPatientFromApi,
        residentCongregateSetting: updatedPatientFromApi.residentCongregateSetting
          ? "YES"
          : "NO",
        employedInHealthcare: updatedPatientFromApi.employedInHealthcare
          ? "YES"
          : "NO",
      })
    );

    setNextPage(true);
  };

  return (
    <PatientTimeOfTestContainer currentPage={"profile"}>
      <PersonForm
        patient={{
          ...patient,
          facilityId: patient.facility === null ? null : patient.facility?.id,
        }}
        activeFacilityId={facility.id}
        patientId={patient.internalId}
        isPxpView={true}
        hideFacilitySelect={true}
        backCallback={() => {
          setPrevPage(true);
        }}
        savePerson={savePerson}
      />
    </PatientTimeOfTestContainer>
  );
};

export default connect()(PatientFormContainer);
