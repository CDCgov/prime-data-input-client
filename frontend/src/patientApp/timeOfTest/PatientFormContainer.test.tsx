import renderer from "react-test-renderer";
import { render, screen, cleanup } from "@testing-library/react";
import { MockedProvider } from "@apollo/client/testing";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

import PatientFormContainer from "./PatientFormContainer";

jest.mock("../..//app/commonComponents/ComboBox", () => () => <></>);
const mockStore = configureStore([]);

jest.mock("react-router-dom", () => ({
  Prompt: (props: any) => <></>,
  useHistory: () => ({
    listen: jest.fn(),
    push: jest.fn(),
  }),
}));

describe("PatientFormContainer", () => {
  afterEach(cleanup);

  it("snapshot", () => {
    const store = mockStore({
      patient: {
        residentCongregateSetting: true,
        employedInHealthcare: true,
        birthDate: "",
      },
      plid: "foo",
      facility: {
        id: "123",
      },
      facilities: [],
    });
    const component = renderer.create(
      <Provider store={store}>
        <MockedProvider mocks={[]} addTypename={false}>
          <PatientFormContainer />
        </MockedProvider>
      </Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
  describe("testing-library/react renderer", () => {
    window.scrollTo = jest.fn();

    beforeEach(() => {
      const store = mockStore({
        patient: {
          residentCongregateSetting: true,
          employedInHealthcare: true,
          birthDate: "",
        },
        plid: "foo",
        facility: {
          id: "123",
        },
        facilities: [],
      });
      render(
        <Provider store={store}>
          <MockedProvider mocks={[]} addTypename={false}>
            <PatientFormContainer />
          </MockedProvider>
        </Provider>
      );
    });
    it("shows required field instruction once", () => {
      expect(
        screen.getByText("All fields marked with", { exact: false })
      ).toBeInTheDocument();
    });
    it("doesn't show a facility input", () => {
      expect(
        screen.queryByLabelText("Facility", {
          exact: false,
        }) as HTMLSelectElement
      ).toBeNull();
    });
  });
});
