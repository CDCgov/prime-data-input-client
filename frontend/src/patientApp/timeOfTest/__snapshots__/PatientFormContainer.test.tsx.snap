// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatientFormContainer snapshot 1`] = `
<main
  className="patient-app patient-app--form padding-bottom-4"
>
  <div
    className="grid-container maxw-tablet"
  >
    <div
      aria-label="progress"
      className="usa-step-indicator usa-step-indicator--counters-sm margin-y-205"
    >
      <ol
        className="usa-step-indicator__segments"
      >
        <li
          aria-current={true}
          className="usa-step-indicator__segment usa-step-indicator__segment--current"
        >
          <span
            className="usa-step-indicator__segment-label"
          >
            Profile information
             
            <span
              className="usa-sr-only"
            >
              completed
            </span>
          </span>
        </li>
        <li
          aria-current={false}
          className="usa-step-indicator__segment"
        >
          <span
            className="usa-step-indicator__segment-label"
          >
            Symptoms and history
             
            <span
              className="usa-sr-only"
            >
              completed
            </span>
          </span>
        </li>
      </ol>
      <div
        className="usa-step-indicator__header"
      >
        <h1
          className="usa-step-indicator__heading"
        >
          <span
            className="usa-step-indicator__heading-counter"
          >
            <span
              className="usa-sr-only"
            >
              Step
            </span>
            <span
              className="usa-step-indicator__current-step margin-right-05"
            >
              1
            </span>
            <span
              className="usa-step-indicator__total-steps"
            >
              of 
              2
            </span>
          </span>
          <span
            className="usa-step-indicator__heading-text"
          >
            Profile information
          </span>
        </h1>
      </div>
    </div>
    <main
      className="prime-edit-patient prime-home padding-top-0"
    >
      <div>
        <p
          className="message--required"
        >
          All fields marked with 
          <span
            className="usa-sr-only"
          >
            Required
          </span>
          <span
            className="usa-hint--required"
          >
             *
          </span>
           are required
        </p>
        <div
          className="prime-formgroup"
        >
          <fieldset
            className="usa-fieldset"
          >
            <legend
              className="prime-formgroup-heading usa-legend"
            >
              General info
            </legend>
            <div
              className="usa-form"
            >
              <div
                className="usa-form-group"
              >
                <label
                  className="usa-label"
                  htmlFor="1"
                >
                  First name
                  <span
                    className="usa-sr-only"
                  >
                    Required
                  </span>
                  <span
                    className="usa-hint--required"
                  >
                     *
                  </span>
                </label>
                <input
                  aria-required={true}
                  className="usa-input"
                  id="1"
                  name="firstName"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </div>
              <div
                className="usa-form-group"
              >
                <label
                  className="usa-label"
                  htmlFor="2"
                >
                  Middle name
                </label>
                <input
                  aria-required="false"
                  className="usa-input"
                  id="2"
                  name="middleName"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </div>
              <div
                className="usa-form-group"
              >
                <label
                  className="usa-label"
                  htmlFor="3"
                >
                  Last name
                  <span
                    className="usa-sr-only"
                  >
                    Required
                  </span>
                  <span
                    className="usa-hint--required"
                  >
                     *
                  </span>
                </label>
                <input
                  aria-required={true}
                  className="usa-input"
                  id="3"
                  name="lastName"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className="usa-form"
            >
              <div
                className="usa-form-group prime-dropdown "
              >
                <label
                  className="usa-label"
                  htmlFor="4"
                >
                  Role
                </label>
                <select
                  aria-required="false"
                  className="usa-select"
                  id="4"
                  name="role"
                  onChange={[Function]}
                  value=""
                >
                  <option
                    value=""
                  >
                    - Select -
                  </option>
                  <option
                    value="STAFF"
                  >
                    Staff
                  </option>
                  <option
                    value="RESIDENT"
                  >
                    Resident
                  </option>
                  <option
                    value="STUDENT"
                  >
                    Student
                  </option>
                  <option
                    value="VISITOR"
                  >
                    Visitor
                  </option>
                </select>
              </div>
            </div>
            <div
              className="usa-form"
            >
              <div
                className="usa-form-group"
              >
                <label
                  className="usa-label"
                  htmlFor="5"
                >
                  Date of birth (mm/dd/yyyy)
                  <span
                    className="usa-sr-only"
                  >
                    Required
                  </span>
                  <span
                    className="usa-hint--required"
                  >
                     *
                  </span>
                </label>
                <input
                  aria-required={true}
                  className="usa-input"
                  id="5"
                  name="birthDate"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="date"
                  value=""
                />
              </div>
            </div>
          </fieldset>
        </div>
        <div
          className="prime-formgroup"
        >
          <fieldset
            className="usa-fieldset"
          >
            <legend
              className="prime-formgroup-heading usa-legend"
            >
              Contact information
            </legend>
            <div
              className="usa-form"
            >
              <div
                className="grid-row grid-gap"
              >
                <div
                  className="mobile-lg:grid-col-6"
                >
                  <div
                    className="usa-form-group"
                  >
                    <label
                      className="usa-label"
                      htmlFor="6"
                    >
                      Phone number
                      <span
                        className="usa-sr-only"
                      >
                        Required
                      </span>
                      <span
                        className="usa-hint--required"
                      >
                         *
                      </span>
                    </label>
                    <input
                      aria-required={true}
                      className="usa-input"
                      id="6"
                      name="telephone"
                      onBlur={[Function]}
                      onChange={[Function]}
                      type="tel"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                className="usa-form-group"
              >
                <label
                  className="usa-label"
                  htmlFor="7"
                >
                  Email address
                </label>
                <input
                  aria-required="false"
                  className="usa-input"
                  id="7"
                  name="email"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="email"
                  value=""
                />
              </div>
            </div>
            <div
              className="usa-form"
            >
              <div
                className="usa-form-group"
              >
                <label
                  className="usa-label"
                  htmlFor="8"
                >
                  Street address 1
                  <span
                    className="usa-sr-only"
                  >
                    Required
                  </span>
                  <span
                    className="usa-hint--required"
                  >
                     *
                  </span>
                </label>
                <input
                  aria-required={true}
                  className="usa-input"
                  id="8"
                  name="street"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className="usa-form"
            >
              <div
                className="usa-form-group"
              >
                <label
                  className="usa-label"
                  htmlFor="9"
                >
                  Street address 2
                </label>
                <input
                  aria-required="false"
                  className="usa-input"
                  id="9"
                  name="streetTwo"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className="usa-form"
            >
              <div
                className="usa-form-group"
              >
                <label
                  className="usa-label"
                  htmlFor="10"
                >
                  City
                </label>
                <input
                  aria-required="false"
                  className="usa-input"
                  id="10"
                  name="city"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </div>
              <div
                className="usa-form-group"
              >
                <label
                  className="usa-label"
                  htmlFor="11"
                >
                  County
                </label>
                <input
                  aria-required="false"
                  className="usa-input"
                  id="11"
                  name="county"
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </div>
              <div
                className="grid-row grid-gap"
              >
                <div
                  className="mobile-lg:grid-col-6"
                >
                  <div
                    className="usa-form-group prime-dropdown "
                  >
                    <label
                      className="usa-label"
                      htmlFor="12"
                    >
                      State
                      <span
                        className="usa-sr-only"
                      >
                        Required
                      </span>
                      <span
                        className="usa-hint--required"
                      >
                         *
                      </span>
                    </label>
                    <select
                      aria-required={true}
                      className="usa-select"
                      id="12"
                      name="state"
                      onBlur={[Function]}
                      onChange={[Function]}
                      value=""
                    >
                      <option
                        value=""
                      >
                        - Select -
                      </option>
                      <option
                        value="AK"
                      >
                        AK
                      </option>
                      <option
                        value="AL"
                      >
                        AL
                      </option>
                      <option
                        value="AR"
                      >
                        AR
                      </option>
                      <option
                        value="AS"
                      >
                        AS
                      </option>
                      <option
                        value="AZ"
                      >
                        AZ
                      </option>
                      <option
                        value="CA"
                      >
                        CA
                      </option>
                      <option
                        value="CO"
                      >
                        CO
                      </option>
                      <option
                        value="CT"
                      >
                        CT
                      </option>
                      <option
                        value="DC"
                      >
                        DC
                      </option>
                      <option
                        value="DE"
                      >
                        DE
                      </option>
                      <option
                        value="FL"
                      >
                        FL
                      </option>
                      <option
                        value="FM"
                      >
                        FM
                      </option>
                      <option
                        value="GA"
                      >
                        GA
                      </option>
                      <option
                        value="GU"
                      >
                        GU
                      </option>
                      <option
                        value="HI"
                      >
                        HI
                      </option>
                      <option
                        value="IA"
                      >
                        IA
                      </option>
                      <option
                        value="ID"
                      >
                        ID
                      </option>
                      <option
                        value="IL"
                      >
                        IL
                      </option>
                      <option
                        value="IN"
                      >
                        IN
                      </option>
                      <option
                        value="KS"
                      >
                        KS
                      </option>
                      <option
                        value="KY"
                      >
                        KY
                      </option>
                      <option
                        value="LA"
                      >
                        LA
                      </option>
                      <option
                        value="MA"
                      >
                        MA
                      </option>
                      <option
                        value="MD"
                      >
                        MD
                      </option>
                      <option
                        value="ME"
                      >
                        ME
                      </option>
                      <option
                        value="MH"
                      >
                        MH
                      </option>
                      <option
                        value="MI"
                      >
                        MI
                      </option>
                      <option
                        value="MN"
                      >
                        MN
                      </option>
                      <option
                        value="MO"
                      >
                        MO
                      </option>
                      <option
                        value="MP"
                      >
                        MP
                      </option>
                      <option
                        value="MS"
                      >
                        MS
                      </option>
                      <option
                        value="MT"
                      >
                        MT
                      </option>
                      <option
                        value="NC"
                      >
                        NC
                      </option>
                      <option
                        value="ND"
                      >
                        ND
                      </option>
                      <option
                        value="NE"
                      >
                        NE
                      </option>
                      <option
                        value="NH"
                      >
                        NH
                      </option>
                      <option
                        value="NJ"
                      >
                        NJ
                      </option>
                      <option
                        value="NM"
                      >
                        NM
                      </option>
                      <option
                        value="NV"
                      >
                        NV
                      </option>
                      <option
                        value="NY"
                      >
                        NY
                      </option>
                      <option
                        value="OH"
                      >
                        OH
                      </option>
                      <option
                        value="OK"
                      >
                        OK
                      </option>
                      <option
                        value="OR"
                      >
                        OR
                      </option>
                      <option
                        value="PA"
                      >
                        PA
                      </option>
                      <option
                        value="PR"
                      >
                        PR
                      </option>
                      <option
                        value="PW"
                      >
                        PW
                      </option>
                      <option
                        value="RI"
                      >
                        RI
                      </option>
                      <option
                        value="SC"
                      >
                        SC
                      </option>
                      <option
                        value="SD"
                      >
                        SD
                      </option>
                      <option
                        value="TN"
                      >
                        TN
                      </option>
                      <option
                        value="TX"
                      >
                        TX
                      </option>
                      <option
                        value="UT"
                      >
                        UT
                      </option>
                      <option
                        value="VA"
                      >
                        VA
                      </option>
                      <option
                        value="VI"
                      >
                        VI
                      </option>
                      <option
                        value="VT"
                      >
                        VT
                      </option>
                      <option
                        value="WA"
                      >
                        WA
                      </option>
                      <option
                        value="WI"
                      >
                        WI
                      </option>
                      <option
                        value="WV"
                      >
                        WV
                      </option>
                      <option
                        value="WY"
                      >
                        WY
                      </option>
                    </select>
                  </div>
                </div>
                <div
                  className="mobile-lg:grid-col-6"
                >
                  <div
                    className="usa-form-group"
                  >
                    <label
                      className="usa-label"
                      htmlFor="13"
                    >
                      Zip code
                      <span
                        className="usa-sr-only"
                      >
                        Required
                      </span>
                      <span
                        className="usa-hint--required"
                      >
                         *
                      </span>
                    </label>
                    <input
                      aria-required={true}
                      className="usa-input"
                      id="13"
                      name="zipCode"
                      onBlur={[Function]}
                      onChange={[Function]}
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </fieldset>
        </div>
        <div
          className="prime-formgroup"
        >
          <fieldset
            className="usa-fieldset"
          >
            <legend
              className="prime-formgroup-heading usa-legend"
            >
              Demographics
            </legend>
            <p
              className="usa-hint maxw-prose"
            >
              This information is important for public health efforts to recognize and address inequality in health outcomes.
            </p>
            <fieldset
              className="usa-fieldset prime-radios"
            >
              <legend
                className="usa-legend"
              >
                Race
              </legend>
              <div
                className="usa-form-group"
              >
                <div
                  className="usa-radio"
                >
                  <input
                    checked={false}
                    className="usa-radio__input"
                    data-required="false"
                    disabled={false}
                    id="14-val-native"
                    name="race"
                    onChange={[Function]}
                    type="radio"
                    value="native"
                  />
                  <label
                    className="usa-radio__label"
                    htmlFor="14-val-native"
                  >
                    American Indian or Alaskan Native
                  </label>
                </div>
                <div
                  className="usa-radio"
                >
                  <input
                    checked={false}
                    className="usa-radio__input"
                    data-required="false"
                    disabled={false}
                    id="14-val-asian"
                    name="race"
                    onChange={[Function]}
                    type="radio"
                    value="asian"
                  />
                  <label
                    className="usa-radio__label"
                    htmlFor="14-val-asian"
                  >
                    Asian
                  </label>
                </div>
                <div
                  className="usa-radio"
                >
                  <input
                    checked={false}
                    className="usa-radio__input"
                    data-required="false"
                    disabled={false}
                    id="14-val-black"
                    name="race"
                    onChange={[Function]}
                    type="radio"
                    value="black"
                  />
                  <label
                    className="usa-radio__label"
                    htmlFor="14-val-black"
                  >
                    Black or African American
                  </label>
                </div>
                <div
                  className="usa-radio"
                >
                  <input
                    checked={false}
                    className="usa-radio__input"
                    data-required="false"
                    disabled={false}
                    id="14-val-pacific"
                    name="race"
                    onChange={[Function]}
                    type="radio"
                    value="pacific"
                  />
                  <label
                    className="usa-radio__label"
                    htmlFor="14-val-pacific"
                  >
                    Native Hawaiian or other Pacific Islander
                  </label>
                </div>
                <div
                  className="usa-radio"
                >
                  <input
                    checked={false}
                    className="usa-radio__input"
                    data-required="false"
                    disabled={false}
                    id="14-val-white"
                    name="race"
                    onChange={[Function]}
                    type="radio"
                    value="white"
                  />
                  <label
                    className="usa-radio__label"
                    htmlFor="14-val-white"
                  >
                    White
                  </label>
                </div>
                <div
                  className="usa-radio"
                >
                  <input
                    checked={false}
                    className="usa-radio__input"
                    data-required="false"
                    disabled={false}
                    id="14-val-unknown"
                    name="race"
                    onChange={[Function]}
                    type="radio"
                    value="unknown"
                  />
                  <label
                    className="usa-radio__label"
                    htmlFor="14-val-unknown"
                  >
                    Unknown
                  </label>
                </div>
                <div
                  className="usa-radio"
                >
                  <input
                    checked={false}
                    className="usa-radio__input"
                    data-required="false"
                    disabled={false}
                    id="14-val-refused"
                    name="race"
                    onChange={[Function]}
                    type="radio"
                    value="refused"
                  />
                  <label
                    className="usa-radio__label"
                    htmlFor="14-val-refused"
                  >
                    Refused to answer
                  </label>
                </div>
              </div>
            </fieldset>
            <fieldset
              className="usa-fieldset prime-radios"
            >
              <legend
                className="usa-legend"
              >
                Ethnicity
              </legend>
              <div
                className="usa-form-group"
              >
                <div
                  className="usa-radio"
                >
                  <input
                    checked={false}
                    className="usa-radio__input"
                    data-required="false"
                    disabled={false}
                    id="15-val-hispanic"
                    name="ethnicity"
                    onChange={[Function]}
                    type="radio"
                    value="hispanic"
                  />
                  <label
                    className="usa-radio__label"
                    htmlFor="15-val-hispanic"
                  >
                    Hispanic or Latino
                  </label>
                </div>
                <div
                  className="usa-radio"
                >
                  <input
                    checked={false}
                    className="usa-radio__input"
                    data-required="false"
                    disabled={false}
                    id="15-val-not_hispanic"
                    name="ethnicity"
                    onChange={[Function]}
                    type="radio"
                    value="not_hispanic"
                  />
                  <label
                    className="usa-radio__label"
                    htmlFor="15-val-not_hispanic"
                  >
                    Not Hispanic
                  </label>
                </div>
              </div>
            </fieldset>
            <fieldset
              className="usa-fieldset prime-radios"
            >
              <legend
                className="usa-legend"
              >
                Biological Sex
              </legend>
              <div
                className="usa-form-group"
              >
                <div
                  className="usa-radio"
                >
                  <input
                    checked={false}
                    className="usa-radio__input"
                    data-required="false"
                    disabled={false}
                    id="16-val-male"
                    name="gender"
                    onChange={[Function]}
                    type="radio"
                    value="male"
                  />
                  <label
                    className="usa-radio__label"
                    htmlFor="16-val-male"
                  >
                    Male
                  </label>
                </div>
                <div
                  className="usa-radio"
                >
                  <input
                    checked={false}
                    className="usa-radio__input"
                    data-required="false"
                    disabled={false}
                    id="16-val-female"
                    name="gender"
                    onChange={[Function]}
                    type="radio"
                    value="female"
                  />
                  <label
                    className="usa-radio__label"
                    htmlFor="16-val-female"
                  >
                    Female
                  </label>
                </div>
                <div
                  className="usa-radio"
                >
                  <input
                    checked={false}
                    className="usa-radio__input"
                    data-required="false"
                    disabled={false}
                    id="16-val-other"
                    name="gender"
                    onChange={[Function]}
                    type="radio"
                    value="other"
                  />
                  <label
                    className="usa-radio__label"
                    htmlFor="16-val-other"
                  >
                    Other
                  </label>
                </div>
              </div>
            </fieldset>
          </fieldset>
        </div>
        <div
          className="prime-formgroup"
        >
          <fieldset
            className="usa-fieldset"
          >
            <legend
              className="prime-formgroup-heading usa-legend"
            >
              Other
            </legend>
            <fieldset
              className="usa-fieldset prime-radios"
            >
              <legend
                className="usa-legend"
              >
                Resident in congregate care/living setting?
                <span
                  className="usa-sr-only"
                >
                  Required
                </span>
                <span
                  className="usa-hint--required"
                >
                   *
                </span>
              </legend>
              <div
                className="usa-form-group"
              >
                <div
                  className="usa-radio"
                >
                  <input
                    checked={true}
                    className="usa-radio__input"
                    data-required={true}
                    disabled={false}
                    id="17-val-YES"
                    name="residentCongregateSetting"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="radio"
                    value="YES"
                  />
                  <label
                    className="usa-radio__label"
                    htmlFor="17-val-YES"
                  >
                    Yes
                  </label>
                </div>
                <div
                  className="usa-radio"
                >
                  <input
                    checked={false}
                    className="usa-radio__input"
                    data-required={true}
                    disabled={false}
                    id="17-val-NO"
                    name="residentCongregateSetting"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="radio"
                    value="NO"
                  />
                  <label
                    className="usa-radio__label"
                    htmlFor="17-val-NO"
                  >
                    No
                  </label>
                </div>
              </div>
            </fieldset>
            <fieldset
              className="usa-fieldset prime-radios"
            >
              <legend
                className="usa-legend"
              >
                Work in Healthcare?
                <span
                  className="usa-sr-only"
                >
                  Required
                </span>
                <span
                  className="usa-hint--required"
                >
                   *
                </span>
              </legend>
              <div
                className="usa-form-group"
              >
                <div
                  className="usa-radio"
                >
                  <input
                    checked={true}
                    className="usa-radio__input"
                    data-required={true}
                    disabled={false}
                    id="18-val-YES"
                    name="employedInHealthcare"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="radio"
                    value="YES"
                  />
                  <label
                    className="usa-radio__label"
                    htmlFor="18-val-YES"
                  >
                    Yes
                  </label>
                </div>
                <div
                  className="usa-radio"
                >
                  <input
                    checked={false}
                    className="usa-radio__input"
                    data-required={true}
                    disabled={false}
                    id="18-val-NO"
                    name="employedInHealthcare"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="radio"
                    value="NO"
                  />
                  <label
                    className="usa-radio__label"
                    htmlFor="18-val-NO"
                  >
                    No
                  </label>
                </div>
              </div>
            </fieldset>
          </fieldset>
        </div>
        <div
          className="mobile-lg:display-flex flex-justify-end margin-top-2"
        >
          <button
            className="usa-button usa-button-disabled"
            disabled={true}
            id="edit-patient-save-lower"
            onClick={[Function]}
            type="button"
          >
            Save and continue
          </button>
          <button
            className="usa-button usa-button--outline margin-top-1 mobile-lg:margin-top-0 margin-right-0"
            onClick={[Function]}
            type="button"
          >
            Back
          </button>
        </div>
      </div>
    </main>
  </div>
</main>
`;
